light_curves/code_src/rubin_functions.py

```diff
@@ -0,0 +1,225 @@
+    token_file = Path.home() / '.rsp-tap.token'    
+    if not os.path.exists(token_file):
+        raise FileNotFoundError(f"Token file not found: {token_file}")
+    with open(token_file, 'r') as f:
+        token = f.readline().strip()
+        if not token:
+            raise FileNotFoundError("No token found in token file.")
+
+    # Build credential and authenticated session
+    cred = pyvo.auth.CredentialStore()
+    cred.set_password("x-oauth-basic", token)
+    session = cred.get("ivo://ivoa.net/sso#BasicAA")
+    if session is None:
+        raise RuntimeError("Failed to obtain authenticated session. Check your token permissions.")
+
+    # Instantiate TAPService
+    rsp_tap_url = 'https://data.lsst.cloud/api/tap'
+    rsp_tap = pyvo.dal.TAPService(rsp_tap_url, session=session)
+    if rsp_tap.baseurl != rsp_tap_url:
+        raise RuntimeError(
+            f"rubin_authenticate: unexpected TAPService.baseurl "
+            f"'{rsp_tap.baseurl}', expected '{rsp_tap_url}'"
+        )
+    return rsp_tap
+
+
+    
+def rubin_get_objectids(sample_table, rsp_tap, search_radius=0.001):
+    """
+    Perform cone searches for each row in sample_table to retrieve matching DP0 object IDs.
+
+    Parameters
+    ----------
+    sample_table : astropy.table.Table
+        Table with columns ['coord', 'objectid', 'label'], where 'coord' is a SkyCoord.
+    rsp_tap : pyvo.dal.TAPService
+        Authenticated TAP service instance returned by `rubin_authenticate()`.
+    search_radius : float, deg
+        Radius of the cone search in degrees (default: 0.001).
+
+    Returns
+    -------
+    combined : astropy.table.Table
+        Combined table with columns ['coord_ra', 'coord_dec', 'objectId',
+        'in_objid', 'in_label'] tagging each DP0 Object match with the input sample info.
+    """
+    #could consider parallelizing this but for now we wait for TAP table upload and real data
+    all_tables = []
+    for row in sample_table:
+        ra = row['coord'].ra.deg
+        dec = row['coord'].dec.deg
+        in_objid = row['objectid']
+        in_label = row['label']
+        
+        query = textwrap.dedent(f"""
+        SELECT coord_ra, coord_dec, objectId
+        FROM dp02_dc2_catalogs.Object
+        WHERE CONTAINS(
+            POINT('ICRS', coord_ra, coord_dec),
+            CIRCLE('ICRS', {ra:.6f}, {dec:.6f}, {search_radius})
+        ) = 1
+        AND detect_isPrimary = 1
+        """).strip()
+        tbl = rsp_tap.run_sync(query).to_table()
+        # skip if no matches
+        if len(tbl) == 0:
+            continue
+        tbl['in_objid'] = in_objid
+        tbl['in_label'] = in_label
+        all_tables.append(tbl)
+
+    #in the case of an empty table, return an empty table
+    if not all_tables:
+        cols = ['coord_ra','coord_dec','objectId','in_objid','in_label']
+        # build a dict mapping each name to an empty list
+        empty_dict = {col: [] for col in cols}
+        return Table(empty_dict)
```

Comment:
Seems simpler to do:

```suggestion
    if not all_tables:
        return
```

Then after calling this in `rubin_get_lightcurves()`, do:

```python
if not obj_table:
    return MultiIndexDFObject()
```

and skip lines 131-138 in `rubin_access_lc_catalog()`.